docs/source/examples/e2e_flow.rst

```diff
@@ -0,0 +1,438 @@
+:ref:`LoRA Tutorial <lora_finetune_label>`.
+
+We'll fine-tune using our
+`single device LoRA recipe <https://github.com/pytorch/torchtune/blob/main/recipes/lora_finetune_single_device.py>`_
+and use the standard settings from the
+`default config <https://github.com/pytorch/torchtune/blob/main/recipes/configs/llama2/7B_lora_single_device.yaml>`_.
+
+This will fine-tune our model on the
+`Alpaca Dataset <https://github.com/pytorch/torchtune/blob/main/torchtune/datasets/_alpaca.py>`_
+using a ``batch_size`` of ``2`` and ``dtype`` of ``bfloat16``. With these settings the model
+should have a peak memory usage of ~16GB and total training time of around 2hours for each epoch.
+We'll need to make some changes to the config to make sure our recipe can access the
+right checkpoints.
+
+Let's first copy over the config to our local working director so we can make changes.
+
+
+.. code-block:: bash
+
+    tune cp llama2/7B_lora_single_device ./custom_lora_config.yaml
+
+
+Let's modify ``custom_lora_config.yaml`` to include the following changes.
+
+.. code-block:: yaml
+
+    checkpointer:
+        # checkpointer to use
+        _component_: torchtune.utils.FullModelHFCheckpointer
+
+        # directory with the checkpoint files
+        # this should match the output_dir above
+        checkpoint_dir: <checkpoint_dir>
+
+        # checkpoint files. For the llama2-7b-hf model we have
+        # 2 .bin files
+        checkpoint_files: [
+            pytorch_model-00001-of-00002.bin,
+            pytorch_model-00002-of-00002.bin,
+        ]
+
+        # since we're starting a new training run, there's no
+        # recipe state and so set this to null
+        recipe_checkpoint: null
+
+        # dir for saving the output checkpoints. Usually set
+        # to be the same as checkpoint_dir
+        output_dir: <checkpoint_dir>
+
+        # model_type which specifies how to convert the state_dict
+        # into a format which TorchTune understands
+        model_type: LLAMA2
+
+    resume_from_checkpoint: False
+
+    # Make sure to update the tokenizer path to the right
+    # checkpoint directory as well
+    tokenizer:
+        _component_: torchtune.models.llama2.llama2_tokenizer
+        path: <checkpoint_dir>/tokenizer.model
+
+
+Once the config is updated, let's kick off training!
+
+
+.. code-block:: bash
+
+    tune run lora_finetune_single_device \
+    --config ./custom_lora_config.yaml
+
+
+Once training is complete, you'll see the following in the logs.
+
+.. code-block:: bash
+
+    [_checkpointer.py:473] Model checkpoint of size 9.98 GB saved to <checkpoint_dir>/hf_model_0001_0.pt
+
+    [_checkpointer.py:473] Model checkpoint of size 3.50 GB saved to <checkpoint_dir>/hf_model_0002_0.pt
+
+    [_checkpointer.py:484] Adapter checkpoint of size 0.01 GB saved to <checkpoint_dir>/adapter_0.pt
+
+
+The "merged weights" (see the :ref:`LoRA Tutorial <lora_finetune_label>` for more details)
```

Comment:
Personally I would avoid the use of quotes here. You could even just say "The final trained weights are merged with the original model and split across two checkpoint files.."